docs/DeveloperGuide.adoc

```diff
@@ -137,76 +146,174 @@ image:BetterModelClassDiagram.png[]
 .Structure of the Storage Component
 image::StorageClassDiagram.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/storage/Storage.java[`Storage.java`]
+*API* : link:{repoURL}/src/main/java/igrad/memo/Storage.java[`Storage.java`]
 
 The `Storage` component,
 
 * can save `UserPref` objects in json format and read it back.
-* can save the Address Book data in json format and read it back.
+* can save the Course Book data in json format and read it back.
 
 [[Design-Commons]]
 === Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the `iGrad.commons` package.
 
 == Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
-// tag::undoredo[]
+=== Adding Modules from NUS Mods
+
+==== Implementation
+
+The automatic filling in of module details on addition of a new module is facilitated by `NusModsRequester`.
+It creates a new instance of `GetRequestManager` in order to make a request to link:https://nusmods.com/[NUSMods].
+Upon receiving a response, it creates an instance of `JsonParsedModule`.
+
+`JsonParsedModule` parses the JSON object given in the response of the initial request and stores the following values:
+
+* `title` e.g. Software Engineering
+* `moduleCode` e.g. CS2103T
+* `credits` e.g. 4
+* `description` e.g. This module introduces the necessary conceptual and analytical tools for systematic and rigorous development of software systems...
+
+The above process is initiated by `AddAutoCommandParser` which then converts a `JsonParsedModule` instance to a `Module`.
+
+The sequence diagram below provide further insight into its execution:
+
+image::AutoAddSequenceDiagram.png[]
+
+As NUSMods might not always have the latest data for the current academic year, a follow up request is made when an empty response is received.
+
+The following activity diagram illustrates this:
+
+image::AutoAddActivityDiagram.png[]
```

Comment:
Is the activity diagram complete?
What if "the previous academic year" request still does not return a valid response? Do we make another API call for previous previous academic year?